single-registry-architecture/README.md

```diff
@@ -0,0 +1,217 @@
+- shut down `crsync` entirely
+- delete Aliyun registry
+
+### Helm charts
+
+The blocker for switching Helm Charts entirely to OCI is currently our artifact discovery process, which uses the
+`index.yaml` file, that is available in HTTP based Helm registries only. So, to get started, we have to switch our
+discovery process to OCI based one. Currently, we're already uploading our build artifacts to both HTTP and OCI repos,
+so nothing is blocking implementation and a switch.
+
+High level migration plan:
+
+- prepare and deploy `distribution` cache instance, configure it for the upstream ACR registry
+  - ensure monitoring and alerting
+- switch our CI/CD pipelines to not tag dev build artifacts
+  - we want to use retention policies for helm charts as well
+- implement changes in the `app-operator`
+  - use OCI discovery instead of `index.yaml` from HTTP registries
+  - (optional) solve the problem of presenting only X (by default 5) most recent versions in the ACE CR
+  - introduce a concept of primary and fall-back registry
+- stop uploading charts into HTTP registries
+- reconfigure `app-operator` to use the OCI cache as the primary registry and upstream ACR as fallback
+- cleanup HTTP registries
+- handle China as for images
+
+## Open questions
+
+Before we start on the implementation, we have to figure out answers to questions below:
+
+1. Should we use a single cache instance per installation, or rather per cloud provider + region?
+
+   The disk requirements for a cache might be significant and as a result cost of running an instance per MC can be quite
+   high in total. Additionally, we can expect many of the images to be the same for all clusters (MCs and WCs) that run
+   for the same provider. So, it seems running a single instance of the cache per cloud provider can be both much cheaper
+   and also more efficient and performant.
+
+1. How to not tag dev builds?
+
+   ACR has a neat new feature that allows for automatic cleanup of untagged artifacts after a certain period. Currently,
+   we upload a ton of dev build images. This feature allows us to clean them up automatically, with zero effort on
+   the cleaning process on our side. Still, we have to switch our CI/CD configs to not tag dev builds in any way. What's
+   the best way to do it?
+
+1. Customers will stop using our Docker Hub subscription.
+
+   Most customers aren't probably even aware of this, but right now all the images the nodes of any cluster pull from
+   Docker Hub, they do so using our Docker Hub subscription. After implementing this change, we wil stop doing this.
+   Moreover, we will ensure unlimited downloads only for images that we manually picked and uploaded (using `retagger`)
+   to our new ACR registry. We have to explain and announce this change to our customers. Also, we probably have to
+   accept that some customers might want us to configure `containerd` to use their Docker Hub credentials to avoid
+   rate limiting. How do we do that?
```

Comment:
Big customers use their own registries and have policies to stick to that. We can ask AEs to check other customers to get an estimation of workloads affected